### PR TITLE
DSD-317: Text component

### PR DESCRIPTION
**Pull request by bigfishdesign13**
Fixes JIRA ticket [DSD-317](https://jira.nypl.org/browse/DSD-317)

## This PR does the following:
- Adds DS `Text` component composed from Chakra's `Text` component.

## How has this been tested?

- Storybook in local build of DS

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] All new and existing tests passed.

### Front End Review:

- [x] View [the example in Storybook](https://pr715-n2ntxygibmjkb7do9j9xot6myog0tto9.tugboat.qa/)

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr715-n2ntxygibmjkb7do9j9xot6myog0tto9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/615cba507207f264b73337fb

**Comment by bigfishdesign13** [comment]
@EdwinGuzman I feel there must be a better way to handle the variants.  What I have put together is very long winded.  Do you know of a way to easily combine variant styles other than writing unique variants for all of the combinations I am trying to offer with this component?

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr715-n2ntxygibmjkb7do9j9xot6myog0tto9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/615cba507207f264b73337fb

File: src/components/Text/Text.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+| ----------------- | ---------- |
+| Added             | `0.25.1`   |
+| Latest            | `0.25.1`   |
+
+<Description of={Text} />
+
+The `Text` component renders a basic `<p>` element. As its name indicates, the `displaySize` prop controls the size of the text rendered in the component.
+
+<Canvas withToolbar>
+  <Story name="Text" args={{}}>
```
Comment (EdwinGuzman) [inline src/components/Text/Text.stories.mdx:42]: If there are no `args`, you can remove the attribute completely. As well as in line 47.

File: src/components/Text/Text.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  parameters={{
+    design: {
+      type: "figma",
+      url:
+        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/?node-id=10733%3A56919",
+    },
+  }}
+  argTypes={{
+    displaySize: { table: { disable: false } },
+    attributes: { table: { disable: true } },
```
Comment (EdwinGuzman) [inline src/components/Text/Text.stories.mdx]: There are no `attributes` or `modifiers` props so they can be removed here.

File: src/components/Text/Text.stories.mdx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Text/Text.stories.mdx:42]: Hmmm the "Controls" still work. I guess if it's blank then all available props will show up and affect the component.

**Comment by EdwinGuzman** [comment]
>  Do you know of a way to easily combine variant styles other than writing unique variants for all of the combinations I am trying to offer with this component?

Yeah, you can pass in additional props to the `useStyleConfig` function besides variant.
Here's the example in `Icon.tsx`:  https://github.com/NYPL/nypl-design-system/pull/712/files#diff-20e991b9d6d8e15a0f13056b053bf010114d2f34c5ff4c5af5fda8b4cf408fefR54
and in `Icon.ts`: https://github.com/NYPL/nypl-design-system/pull/712/files#diff-1f23f4c65cb8d9a3808487fc16457807ea306e5c8ec86834eedf9ef3fc0703ddR53

You can pass in any prop name and use it in `baseStyle`. So in `Text.tsx`, you can do something like
```
const styles = useStyleConfig("Text", { variant, bold: isBold });`
```

And in `text.ts`:
```
const Text = {
  baseStyle: (props) => ({
    fontWeight: props.isBold ? "bold" : null,
  }),
};
```
This is a very simple example. I see that there are different values for `fontWeight` in `text.ts`... so maybe you will have a some if/else statements. I'm trying to figure this out for the `Card` component right now and it's confusing because there are a lot of cases. But try it and maybe it'll be easier to think about and not as long-winded.

Edit: you can also add more in `baseStyles` for the other values in `props`:
```
baseStyles: (props) => {
  //  more code here
  return {
    fontWeight: ....
  };
},
```

**Comment by bigfishdesign13** [comment]
@EdwinGuzman I cleaned things up and simplified the style object.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr715-n2ntxygibmjkb7do9j9xot6myog0tto9.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/615cba507207f264b73337fb

File: src/components/Text/Text.tsx [hunk 3]
```diff
@@ -31,11 +31,13 @@ function Text(props: React.PropsWithChildren<TextProps>) {
...
-  isBold && (variant = `${variant}Bold`);
-  isItalic && (variant = `${variant}Italic`);
 
-  const styles = useStyleConfig("Text", { variant });
-  const finalClassName = noSpace ? `${className} no-space` : className;
+  const styles = useStyleConfig("Text", {
+    variant,
+    isBold: isBold,
+    isItalic: isItalic,
+    noSpace: noSpace,
```
Comment (EdwinGuzman) [inline src/components/Text/Text.tsx]: You can use the shorthand if you want
```
{
variant,
isBold,
isItalic,
noSpace,
}
```

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**